connector/main.go

```diff
@@ -126,115 +125,74 @@ func main() {
 	}
 
 	// for each course, we will query through the professors in the professors collection
-	professorSet := make(map[string]bool, 10000)
+	// professorSet := make(map[string]bool, 10000
 	for cursor.Next(ctx) {
-		var section uwomodel.Section
-		cursor.Decode(&section)
-		sectionData := section.SectionData
-		professors := make([]string, 0)
-		professorsString := sectionData.Instructor
-
-		// the courseString is the course name in the form of "FACULTY NUMBER+SUFFIX" (e.g. "CS 1350A")
-		courseString := section.CourseData.Faculty + " " + strconv.Itoa(section.CourseData.Number) + section.CourseData.Suffix
-
-		currentString := ""
-		if (professorsString) == "." {
-			continue
-		}
-
-		// iterate through each character in the string
-		for i := 0; i < len(professorsString); i++ {
-			if professorsString[i] == '.' && i != 1 {
-				// add the current string not including the last two characters (the space and first initial)
-				professor := currentString[:len(currentString)-2]
-				if professorSet[professor] == false {
-					professorSet[professor] = true
-					professors = append(professors, professor)
+		var course uwomodel.Course
+		cursor.Decode(&course)
+
+		for _, section := range course.SectionData {
+			// the courseString is the course name in the form of "FACULTY NUMBER+SUFFIX" (e.g. "CS 1350A")
+			courseString := course.CourseData.Faculty + " " + (course.CourseData.Number) + course.CourseData.Suffix
+			for _, professor := range section.Instructors {
+				var rmpProfessor model.MongoProfessor
+				var finalProfessor uwomodel.Professor
+				var existingProfessor uwomodel.Professor
+				firstInitial := professor[0:1]
+				restOfName := professor[3:]
+
+				// query for the professor in the professors collection with the first initial and the rest of the name
+				// use this regexp: ^A. X$, where A=firstInitial, X=restOfName
+				regexString := "^" + firstInitial + "." + "*" + restOfName + "$"
```

Comment:
think we should change the regex string to 
```suggestion
				regexString := "^" + firstInitial + "." + "*" + restOfName + "*" + "$"
```

Reply:
makes the search more inclusive for peeps who have more stuff after their last name on RMP, but idk if it breaks anything or will make the search too forgiving